File: Common/SponsorLevel.cs
using System.Runtime.Serialization;

namespace DevSpace.Common {
	[DataContract]
	public class SponsorLevel : ISponsorLevel {
		private SponsorLevel() {}

		[DataMember]public int Id { get; private set; }
		[DataMember]public int DisplayOrder { get; private set; }
		[DataMember]public string DisplayName { get; private set; }
		[DataMember]public bool DisplayInSidebar { get; private set; }
		[DataMember]public bool DisplayInEmails { get; private set; }
	}
}

File: FSharp.Common/SponsorLevel.fs
namespace DevSpace.FSharp.Common

type SponsorLevel =
  {
    Id:int;
    DisplayOrder:int;
    DisplayName:string;
    DisplayInSidebar:bool;
    DisplayInEmails:bool
  }

  member this.WithId( value ) =
    { this with Id = value }

  member this.WithDisplayOrder( value ) =
    { this with DisplayOrder = value }

  member this.WithDisplayName( value ) =
    {this with DisplayName = value }

  member this.WithDisplayInSidebar( value ) =
    { this with DisplayInSidebar = value }

  member this.WithDisplayInEmails( value ) =
    { this with DisplayInEmails = value }
